rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/Transaction.java

```diff
@@ -690,10 +697,21 @@ public boolean forgetParticipant(String participantUrl) {
         return findLRAParticipant(participantUrl, true) != null;
     }
 
-    public void forgetAllParticipants() {
-        if (pending != null) {
-            pending.forEach(LRARecord::forget);
+    public boolean forgetAllParticipants() {
+        boolean notifiedAll = true;
+
+        Iterator<LRARecord> iterator = pending.iterator();
```

Comment:
just two ideas
* could be `pending` `null` or it could not be?
* I would, personally and if I don't miss some detail, consider more readable the lamda style like `pending.removeIf(record -> record.forget())`